fossilize.cpp

```diff
@@ -7402,7 +7402,8 @@ static bool json_value(const VkDescriptorSetLayoutCreateInfo& layout, Allocator&
 		binding.AddMember("descriptorCount", b.descriptorCount, alloc);
 		binding.AddMember("stageFlags", b.stageFlags, alloc);
 		binding.AddMember("binding", b.binding, alloc);
-		if (b.pImmutableSamplers)
+		if (b.pImmutableSamplers && b.descriptorCount > 0 &&
+		    (b.descriptorType == VK_DESCRIPTOR_TYPE_SAMPLER || b.descriptorType == VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER))
```

Comment:
It makes more sense to fix copy_descriptor_set_layout() instead. There, we forget to clear pImmutableSamplers to nullptr if we hit this scenario.

Reply:
Yeah, it's much better.